tuist/tuist
# Fix parsing of SwiftPackageManager dependencies including `macro` targets

Resolves https://github.com/tuist/tuist/issues/5307

### Short description 📝

Add a new enum case in `TargetType`, and make necessary changes to compile `TuistKit`.

### How to test the changes locally 🧐

Not sure. Didn't see any tests in `TuistSupportTests` relates to parsing Package.swift, and not sure how to add one

### Contributor checklist ✅


### Reviewer checklist ✅
